cmd/pinsharesvc/process.go

```diff
@@ -0,0 +1,386 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"time"
+
+	"golang.org/x/sys/windows/svc/debug"
+)
+
+type ProcessManager struct {
+	config          *ServiceConfig
+	eventLog        debug.Log
+	ipfsCmd         *exec.Cmd
+	pinshareCmd     *exec.Cmd
+	ipfsLogFile     *os.File
+	pinshareLogFile *os.File
+	ipfsExited      chan struct{} // closed when IPFS process exits
+	pinshareExited  chan struct{} // closed when PinShare process exits
+	mu              sync.Mutex
+}
+
+func NewProcessManager(config *ServiceConfig, eventLog debug.Log) *ProcessManager {
+	return &ProcessManager{
+		config:   config,
+		eventLog: eventLog,
+	}
+}
+
+// StartIPFS starts the IPFS daemon
+func (pm *ProcessManager) StartIPFS(ctx context.Context) error {
+	pm.mu.Lock()
+	defer pm.mu.Unlock()
+
+	// Check if IPFS binary exists
+	if _, err := os.Stat(pm.config.IPFSBinary); os.IsNotExist(err) {
+		return fmt.Errorf("IPFS binary not found at %s", pm.config.IPFSBinary)
+	}
+
+	// Initialize IPFS repo if it doesn't exist
+	repoPath := pm.config.GetIPFSRepoPath()
+	if _, err := os.Stat(filepath.Join(repoPath, "config")); os.IsNotExist(err) {
+		pm.logInfo("Initializing IPFS repository...")
+		if err := pm.initializeIPFS(); err != nil {
+			return fmt.Errorf("failed to initialize IPFS: %w", err)
+		}
+	}
+
+	// Open log file
+	logPath := filepath.Join(pm.config.DataDirectory, "logs", "ipfs.log")
+	logFile, err := os.OpenFile(logPath, os.O_CREATE|os.O_APPEND|os.O_WRONLY, 0644)
+	if err != nil {
+		return fmt.Errorf("failed to open IPFS log file: %w", err)
+	}
+	pm.ipfsLogFile = logFile
+
+	// Create command
+	pm.ipfsCmd = exec.CommandContext(ctx, pm.config.IPFSBinary, "daemon")
+	pm.ipfsCmd.Env = append(os.Environ(),
+		fmt.Sprintf("IPFS_PATH=%s", repoPath),
+	)
+	pm.ipfsCmd.Stdout = logFile
+	pm.ipfsCmd.Stderr = logFile
+
+	// Set process group for proper cleanup on Windows
+	pm.ipfsCmd.SysProcAttr = &syscall.SysProcAttr{
+		CreationFlags: syscall.CREATE_NEW_PROCESS_GROUP,
+	}
+
+	// Start the process
+	if err := pm.ipfsCmd.Start(); err != nil {
+		logFile.Close()
+		return fmt.Errorf("failed to start IPFS daemon: %w", err)
+	}
+
+	pm.logInfo(fmt.Sprintf("IPFS daemon started with PID %d", pm.ipfsCmd.Process.Pid))
+
+	// Create exit channel and monitor process in background
+	pm.ipfsExited = make(chan struct{})
+	go pm.monitorProcess(ctx, pm.ipfsCmd, "IPFS", pm.ipfsExited)
+
+	return nil
+}
+
+// initializeIPFS initializes a new IPFS repository
+func (pm *ProcessManager) initializeIPFS() error {
+	repoPath := pm.config.GetIPFSRepoPath()
+
+	cmd := exec.Command(pm.config.IPFSBinary, "init")
+	cmd.Env = append(os.Environ(),
+		fmt.Sprintf("IPFS_PATH=%s", repoPath),
+	)
+
+	output, err := cmd.CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("ipfs init failed: %w\nOutput: %s", err, string(output))
+	}
+
+	pm.logInfo("IPFS repository initialized successfully")
```

Comment:
Again, no need to use the word "repository"

```suggestion
	pm.logInfo("IPFS initialized successfully")
```